src/Symfony/Component/JsonStreamer/Read/PhpGenerator.php

```diff
@@ -58,7 +58,7 @@ public function generate(DataModelNodeInterface $dataModel, bool $decodeFromStre
                 .$providers
                 .($this->canBeDecodedWithJsonDecode($dataModel, $decodeFromStream)
                     ? $this->line('    return \\'.Decoder::class.'::decodeStream($stream, 0, null);', $context)
-                    : $this->line('    return $providers[\''.$dataModel->getIdentifier().'\']($stream, 0, null);', $context))
+                    : $this->line('    return $providers[\''.addcslashes($dataModel->getIdentifier(), "'").'\']($stream, 0, null);', $context))
```

Comment:
We should factorize this repetitive code to make it more readable.:

```php
private static function quote(string $string): string
{
    return \''.addcslashes($string, "'").'\';
}
```

```suggestion
                    : $this->line('    return $providers[self::quote($dataModel->getIdentifier())]($stream, 0, null);', $context))
```

Reply:
Or using `var_export($string, true)` would cover other edge cases (the null char: `\0`, see [PHP sources](https://github.com/php/php-src/blob/c27368c9392c46047c20d86f536753c29e73b8a0/ext/standard/var.c#L543-L545)).